app/services/forest_liana/search_query_builder.rb

```diff
@@ -81,15 +84,24 @@ def search_param
 
         if (@params['searchExtended'].to_i == 1)
           SchemaUtils.one_associations(@resource).map(&:name).each do |association|
+            if @collection.search_fields
+              association_search = @collection.search_fields.map do |field|
+                if field.include?('.') && field.split('.')[0] == association.to_s
+                  field.split('.')[1]
+                end
+              end
```

Comment:
here, again, I am sure that the resulting array `association_search` contains `nil` values that should not be here for performance and code quality (a `compact` call should do the job)